libs/remote_storage/src/lib.rs

```diff
@@ -154,7 +175,21 @@ pub trait RemoteStorage: Send + Sync + 'static {
     /// whereas,
     /// list_prefixes("foo/bar/") = ["cat", "dog"]
     /// See `test_real_s3.rs` for more details.
-    async fn list_files(&self, folder: Option<&RemotePath>) -> anyhow::Result<Vec<RemotePath>>;
+    async fn list_files(&self, prefix: Option<&RemotePath>) -> anyhow::Result<Vec<RemotePath>> {
+        let result = self.list(prefix, ListingMode::NoDelimiter).await?.keys;
+        Ok(result)
+    }
+
+    async fn list(
+        &self,
+        prefix: Option<&RemotePath>,
+        _mode: ListingMode,
+    ) -> anyhow::Result<Listing, DownloadError>; /* {
+                                                     // XXX Placeholder impl.
+                                                     let mut result = Listing::default();
+                                                     result.prefixes = self.list_prefixes(prefix).await?;
+                                                     Ok(result)
+                                                 }*/
```

Comment:
At a minimum, please format the XXX comment more nicely.

Also, I would love some basic tests for this that are run for each impl of this interface.

Reply:
Oops, this was commented out while I was refactoring.  I've just removed it now.

Reply:
In 74d9fcde6888191de29a083daffd2a1cc66e1048 I've added a basic test for LocalFS, just so that we have something that tracks the expected behavior.  I don't know that we have infrastructure for unit testing Azure+S3 code: seems like we rely on the coverage we get in integration tests.